### PR TITLE
Sv pure bypass, Fake latency and improvements to backtrack

### PR DESCRIPTION
**Pull request by notgoodusename**
Useful for extra backtrack and custom sounds and gloves

### CODE REVIEW

**Comment by aljawary** [comment]
custom skybox also?

**Comment by notgoodusename** [comment]
> custom skybox also?

i guess but the old sounds are working fine

**Comment by RyDeem** [comment]
Does this work in casual? i got it working in the aim_botz map but in casual it bugs out and suppresses even kilfeed from showing

**Comment by notgoodusename** [comment]
> Does this work in casual? i got it working in the aim_botz map but in casual it bugs out and suppresses even kilfeed from showing

yes, it does, maybe yours was bugging because you had the original call static, this will cause problems because when you rehook netchan the original call wont change

**Comment by notgoodusename** [comment]
i improved the backtrack tick choosing, and its really fun xd

**Comment by newcommerdontblame** [comment]
Backtrack tickrate chams?

**Comment by Evilhelder** [comment]
Can you extend the backtracking up to 600 ms?

**Comment by notgoodusename** [comment]
> Can you extend the backtracking up to 600 ms?

no, sv_maxunlag prevents that before it was set to 1 now its set to 0,2 if you to learn why you can read this
https://github.com/perilouswithadollarsign/cstrike15_src/blob/master/game/server/player_lagcompensation.cpp#L269-L297

**Comment by Evilhelder** [comment]
> > Can you extend the backtracking up to 600 ms?
> 
> no, sv_maxunlag prevents that before it was set to 1 now its set to 0,2 if you to learn why you can read this
> https://github.com/perilouswithadollarsign/cstrike15_src/blob/master/game/server/player_lagcompensation.cpp#L269-L297

got it!

**Comment by Oly-UK** [comment]
notgoodusername back at it with the best prs!!! thanks for this <3

**Comment by notgoodusename** [comment]
> notgoodusername back at it with the best prs!!! thanks for this <3

Thanks!

**Comment by HackerGeo-sp1ne** [comment]
@notgoodusename what is your discord, i want to speak with you

**Comment by notgoodusename** [comment]
> @notgoodusename what is your discord, i want to speak with you

send me yours and ill add

**Comment by barbadully** [comment]
does it show the extended backtrack chams or ?

**Comment by notgoodusename** [comment]
> does it show the extended backtrack chams or ?

yes it does

**Comment by notgoodusename** [comment]
small clip:
https://streamable.com/u43qpb

**Comment by brutus-star** [comment]
yo. how are you getting custom stuff into csgo, because i was using migi and the menu would never show up

**Comment by RyDeem** [comment]
now just to make aimbot aim at backtrack hitboxes like normal ones...
xd but fr

**Comment by aljawary** [comment]
not working after i add new hooks method :(

**Comment by notgoodusename** [comment]
on hooks.h instead of vmt put HookType

**Comment by aljawary** [comment]
> on hooks.h instead of "Vmt" put "HookType"

still doesnt work for me 😕

**Comment by notgoodusename** [comment]
> > on hooks.h instead of "Vmt" put "HookType"
> 
> still doesnt work for me 😕

found the fix
below 
hooks->networkChannel.hookAt(46, SendDatagram);
paste this
MH_EnableHook((*reinterpret_cast<void***>(network))[40]);
MH_EnableHook((*reinterpret_cast<void***>(network))[46]);

**Comment by aljawary** [comment]
> > > on hooks.h instead of "Vmt" put "HookType"
> > 
> > 
> > still doesnt work for me 😕
> 
> found the fix
> below
> `hooks->networkChannel.hookAt(46, SendDatagram);`
> paste this
> 
> ```
> MH_EnableHook((*reinterpret_cast<void***>(network))[40]);
> MH_EnableHook((*reinterpret_cast<void***>(network))[46]);
> ```

thx fixed 💯

**Comment by AntonDoil** [comment]
> > > > on hooks.h instead of "Vmt" put "HookType"
> > > 
> > > 
> > > still doesnt work for me confused
> > 
> > 
> > found the fix
> > below
> > `hooks->networkChannel.hookAt(46, SendDatagram);`
> > paste this
> > ```
> > MH_EnableHook((*reinterpret_cast<void***>(network))[40]);
> > MH_EnableHook((*reinterpret_cast<void***>(network))[46]);
> > ```
> 
> thx fixed, but my game crash when loading on next match, i need to unhook and reinject again in lobby to prevent from crashing

Same.

File: Osiris/Hooks.cpp [hunk 1]
```diff
@@ -181,19 +181,18 @@ static bool __stdcall createMove(float inputSampleTime, UserCmd* cmd) noexcept
...
+    if (oldPointer != network && network && localPlayer)
     {
         oldPointer = network;
+        Backtrack::UpdateIncomingSequences(true);
         hooks->networkChannel.init(network);
         hooks->networkChannel.hookAt(40, SendNetMsg);
         hooks->networkChannel.hookAt(46, SendDatagram);
-        Backtrack::UpdateIncomingSequences(true);
     }
+    Backtrack::UpdateIncomingSequences():
```
Comment (aljawary) [inline Osiris/Hooks.cpp:195]: should end with ;

Comment (notgoodusename) [reply Osiris/Hooks.cpp:195]: im on phone :d srry

Comment (aljawary) [reply Osiris/Hooks.cpp]: hehe good work man repp+

**Comment by AntonDoil** [comment]
Is possible to update it for actually hooking?

File: Osiris/Hooks.cpp (hunk 1 above)
Comment (aljawary) [reply Osiris/Hooks.cpp]: still crash man :(

**Comment by notgoodusename** [comment]
> Is possible to update it for actually hooking?

maybe

**Comment by AntonDoil** [comment]
still crashin with newest hook

**Comment by notgoodusename** [comment]
the problem is that minhook detects it as already created

File: Osiris/Hooks.cpp (hunk 1 above)
Comment (AntonDoil) [reply Osiris/Hooks.cpp]: > still crashing bruh

yeah ;d

**Comment by AntonDoil** [comment]
> the problem is that minhook detects it as already created

yeah thats why we must hook only 1 time, that what i sent you on the discord Meinkhampf.

**Comment by notgoodusename** [comment]
i know but netchannel needs to be rehooked

**Comment by notgoodusename** [comment]
Doing another pull for the current osiris version

**Comment by wirebruh** [comment]
It worked for me yesterday after updating

**Comment by wirebruh** [comment]
imma test it again rn if it works for me

**Comment by wirebruh** [comment]
i remember it was crashing when shooting a player with extended backtrack enabled but i fixed it idk how

**Comment by wirebruh** [comment]
@notgoodusename yeah, it still works. add me on discord cableman#6383

**Comment by notgoodusename** [comment]
> @notgoodusename yeah, it still works. add me on discord cableman#6383

done

**Comment by RyDeem** [comment]
yaal updating this bois?

**Comment by RyDeem** [comment]
i fixed it.
VmtSwap networkChannel;
in hooks.h

or check

**Comment by RyDeem** [comment]
Id say do the VmtSwap and remove the sv pure bypass from this pull, and reopen it.
One feature at a time is better to get added. Good luck and nice job on the code anyway.

**Pull request closed without merging**